hibernate-core/src/main/java/org/hibernate/engine/internal/StatefulPersistenceContext.java

```diff
@@ -743,7 +743,9 @@ public Object narrowProxy(Object proxy, EntityPersister persister, EntityKey key
 
 			if ( object != null ) {
 				final LazyInitializer li = ( (HibernateProxy) proxy ).getHibernateLazyInitializer();
-				li.setImplementation( object );
+				if ( li.isUninitialized() ) {
```

Comment:
Should it even be possible that this is already initialized? Maybe this should be an assert and we should fix the callers such that the state is set on the existing entity object rather than creating a new one and ignore it here?

Reply:
ok based on the changes to Initializers the proxy should be always initialised, I add an assertion here